kroxylicious-kms-provider-hashicorp-vault/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/StringKekid.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.hashicorp.vault;
+
+import io.kroxylicious.kms.service.KekId;
+import io.kroxylicious.kms.service.KmsException;
+
+public record StringKekid(String keyId) implements KekId {
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <K> K getId(Class<K> keyType) {
+        if (!keyType.isAssignableFrom(String.class)) {
+            throw new KmsException("Unsupported keyType (" + keyType + ") requested. Only Strings are supported");
+        }
+        return (K) this.keyId;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Why the override?

Reply:
resolved in 5a11ea4e88cf4219528390d5b32788a5a1c34807